Pcap++/src/PcapLiveDevice.cpp

```diff
@@ -1055,31 +1056,90 @@ void PcapLiveDevice::setDefaultGateway()
 		}
 	}
 #elif defined(__APPLE__) || defined(__FreeBSD__)
-	std::string command = "netstat -nr | grep default | grep " + m_Name;
-	std::string ifaceInfo = executeShellCommand(command);
-	if (ifaceInfo == "")
-	{
-		PCPP_LOG_DEBUG("Error retrieving default gateway address: couldn't get netstat output");
+
+	//route message struct for communication in BSD / APPLE device
+	struct BSDRoutingMessage{
+		struct	rt_msghdr header;
+		char	messageSpace[512];
```

Comment:
wondering if  `512` is a proper number?

Reply:
The number used is consistent with the one in the macOS source code. I believe it would be safer to retain it as is.